src/back/glsl/mod.rs

```diff
@@ -521,7 +521,11 @@ impl<'a, W: Write> Writer<'a, W> {
                 // glsl has no concept of samplers so we just ignore it
                 TypeInner::Sampler { .. } => continue,
                 // All other globals are written by `write_global`
-                _ => self.write_global(handle, global)?,
+                _ => {
+                    let end_with_newline =
+                        index != self.module.global_variables.len().saturating_sub(1);
```

Comment:
This isn't going to work, since we are skipping the unused globals anyway. So your item may end up being the last in the list.